### ISSUE TITLE
Security lists: icmp_options{} problem

### ISSUE THREAD

**Issue by rramge** [terraform-providers/terraform-provider-oci#784]
### Terraform Version and Provider Version

Terraform v0.12.0
+ provider.oci v3.27.0

### Affected Resource(s)

oci_core_security_list

### Terraform Configuration Files

resource "oci_core_security_list" "SL_10_3_255_0" {
  provider       = "oci.lhr"
  compartment_id = "${var.subtenancy_ocid}"
  display_name   = "SL_10_3_255_0"
  vcn_id         = "${oci_core_virtual_network.VCN_10_3_0_0.id}"
[...]
  ingress_security_rules {
      protocol  = 1
      source    = "10.0.0.0/8"
      stateless = true

      icmp_options {
        "type" = 3
        "code" = 4
      }
  }
[...]
}

### Expected Behavior

Security rule being created.

### Actual Behavior

Error: Missing required argument

  on tenants/ict.technology/lhr/11-securitylist_10.3.255.0.tf line 85, in resource "oci_core_security_list" "SL_10_3_255_0":
  85:       icmp_options {

The argument "type" is required, but no definition was found.

### Steps to Reproduce

1. `terraform validate`

### Important Factoids

This affects all occurences of an icmp_options block in all security lists (tcp_options and udp_options work fine).

After commenting out the icmp_options block, the rules are validated just fine.

The definitions worked fine with terraform 0.11.x, I run into this issue during the migration to terraform 0.12 block syntax, using the security_lists example as orientation.

Thanks,

Ralf

**Comment by varmax2511** [comment]
Hi @rramge, with Terraform  v0.12 `"` is not allowed when specifying arguments in config. Please try the below change for specifying `icmp_options` and let us know if the issue still persists.

`icmp_options {`
`    type = 3`    
`    code = 4`    
 `}`

**Comment by rramge** [comment]
Hi,

you are absolutely right, and I feel absolutely stupid now. Thanks for debugging my crap...

Cheers,

Ralf

**Comment by rramge** [comment]
Hi,

I first thought the issue is resolved, but it isn't.

Quote from the config:

//------- ICMP required rules ---------------------------------------------

  ingress_security_rules {
      protocol  = 1
      source    = "10.0.0.0/8"
      stateless = true

      icmp_options {
        type = 3
        code = 4
      }
    }

  ingress_security_rules {
      protocol  = 1
      source    = "10.0.0.0/8"
      stateless = true

      icmp_options {
        type = 8
        code = 0
      }
    }

  ingress_security_rules {
      protocol  = 1
      source    = "10.0.0.0/8"
      stateless = true

      icmp_options {
        type = 0
        code = 0
      }
    }


Result:

rror: Missing required argument

  on tenants/ict.technology/fra/11-securitylist_10.4.255.0.tf line 45, in resource "oci_core_security_list" "SL_10_4_255_0":
  45:       icmp_options {

The argument "type" is required, but no definition was found.


Error: Missing required argument

  on tenants/ict.technology/fra/11-securitylist_10.4.255.0.tf line 55, in resource "oci_core_security_list" "SL_10_4_255_0":
  55:       icmp_options {

The argument "type" is required, but no definition was found.


Error: Missing required argument

  on tenants/ict.technology/fra/11-securitylist_10.4.255.0.tf line 100, in resource "oci_core_security_list" "SL_10_4_255_0":
 100:       icmp_options {

The argument "type" is required, but no definition was found.

**Comment by varmax2511** [comment]
@rramge I am unable to reproduce the issue using the  `ingress_security_rules` configuration specified above. I would suggest you to take backup of your current config and then run `terraform 0.12upgrade`. This command upgrades config in the current working directory per v0.12 standard.

To read more on `terraform 0.12upgrade` refer https://www.terraform.io/docs/commands/0.12upgrade.html

If the issue still persists, please share your full terraform config (without sensitive information) so we can try replicating the scenario.

**Comment by rramge** [comment]
Hi,

I think I tracked it down and I am quite sure the root cause is related to Terraform, not the oci provider.

My findings tell me that it's indeed related to quotes - but not in the actual lines which are mentioned in the error message. Instead, it happens here with every occurence of icmp_options{} if there arfe quotes in a different line which is processed earlier, e.g. in a different .tf file.

Long story short: I think Terraform is the culprit, not the provider.
